include/alpaka/block/shared/st/BlockSharedMemStMember.hpp

```diff
@@ -29,7 +29,9 @@ namespace alpaka
         public:
             //-----------------------------------------------------------------------------
 #ifndef NDEBUG
-            BlockSharedMemStMemberImpl(uint8_t* mem, std::size_t capacity) : m_mem(mem), m_capacity(capacity)
+            BlockSharedMemStMemberImpl(uint8_t* mem, std::size_t capacity) :
```

Comment:
This call also happens on GPU and thus technically also wastes registers with the 64-bit value. However, this call happens at the beginning of the kernel in a non-thread-parallel section before the payload logic, where probably not many registers are used and thus will likely not increase the overall footprint of non-tiny kernels.

So, I can live with 64-bit here if it serves to simplify the interface.

Reply:
I agree if it makes the interfaces simpler and avoids a lot of cast usage keep it 64bit. This part will not effect the register usage. 
It is only active in the debug build and it consumes only a temporary register at the beginning of the kernel.

Reply:
Okay, so this one I will keep as is.